flink-kubernetes/src/test/java/org/apache/flink/kubernetes/kubeclient/factory/KubernetesJobManagerFactoryTest.java

```diff
@@ -139,6 +140,12 @@ public void testDeploymentMetadata() throws IOException {
         expectedLabels.putAll(userLabels);
         assertEquals(expectedLabels, resultDeployment.getMetadata().getLabels());
 
+        for (Map.Entry<String, String> entry : userAnnotations.entrySet()) {
```

Comment:
Maybe we could use `assertThat(resultDeployment.getMetadata().getAnnotations(), equalTo(userAnnotations));` instead.